### PR TITLE
Namespace refactor work

### PR DESCRIPTION
**Pull request by DDReaper**
Overview
---
Overhaul of namespaces / classes within the project, e.g. 
Rename HoloToolkit folders -> MixedRealityToolkit verbiage.
Rename namespaces like HoloToolkit.Unity to:
MixedRealityToolkit.Input.
MixedRealityToolkit.Common
MixedRealityToolkit.SpatialMapping etc.

Tasks
---
[x] Rename Core HT -> MRTK folder
[x] Rename Examples HT -> Examples MRTK folder
[x] Rename Unit Tests HT -> Unit Tests MRTK folder
[x] Initial global NS rename
[x] Review Input namespace
[x] Review Spacial Mapping namespace
[x] Review boundary namespace
[x] Review Sharing namespace
[x] Review Spacial mapping namespace
[x] Review Spacial Sound namespace
[x] Review Spacial Understanding namespace<- Needs review
[x] Renamed SpacialUnderstanding class to SpacialUnderstandingManager (resolves new namespace conflicts)
[x] Review Utilities
[x] Review Build code
[x] UX code <- needs discussion
[x] Examples code and structure

Changes
---
- Fixes: #1527

PR ready for review.  Should not be merged until the base MRTK Dll's have been updated and included with the same NameSpace change.

### CODE REVIEW

**Comment by StephenHodgson** [comment]
@DDReaper were you refactoring from the dev branch?

I re-targeted the PR and it seems like there's some conflicts.

**Comment by DDReaper** [comment]
Possibly not, I may have started from the master branch. Will review conflicts

**Comment by DDReaper** [comment]
Back to review @StephenHodgson  / @NeerajW  :D

**Comment by DDReaper** [comment]
OK, fixed the latest Conflicts.  Guessing this needs to be merged quickly as new PR's are causing new issues with the rename.
Shouldn't accept new PR's against the dev branch until this one is merged

**Comment by StephenHodgson** [comment]
@DDReaper add the pending review label back once you're done sorting this out.

**Comment by DDReaper** [comment]
Ok, last merge changes have been resolved.  Two open queries for reviewers:
* Common and Utilities got merged in to the Utilities folder. There is no practical reason for these to be split as it doesn't add anything to the project.  However. These are all named in the MRTK.Common namespace.
Should the files in the Utilities folder be in the Common or Utilities namespace? (or rename the entire folder to match the namespace, either way, ONE folder, ONE namespace)

* There are two materials still named HoloToolKit (and I don't have the energy at 2am to wander the project to fix them)
 - MixedRealityToolkit\Utilities\Materials\HoloToolkit_Default.mat
 - MixedRealityToolkit-Examples\Input\Materials\HoloToolkit_Focusable.mat
If someone else wants to pick those up after merge, please do.

Now ready for review

*NOTE*
Polite request, no more merges in to dev until this is merged. Getting kinda fed up fixing the same files over and over again!

**Comment by NeerajW** [comment]
@DDReaper thank you for taking on the mammoth namespace refactor work.
I'll try my best to explain below why this PR is not in a reviewable / acceptable state currently and address some comments.

>>Shouldn't accept new PR's against the dev branch until this one is merged

The right way to solve such big feature work which may take longer to merge is by creating a feature branch. Something like: Feature_NamespaceRefactor and then we can take time to iterate and understand changes. Holding a branch hostage and not accepting PRs is not an acceptable or correct solution.

>>Common and Utilities got merged in to the Utilities folder. There is no practical reason for these to be split as it doesn't add anything to the project. However. These are all named in the MRTK.Common namespace.

I did a bunch of work over the holidays to split Utilities and Common. We should not just merge them back without understanding the reasoning. 

We need the ability to create simpler feature folder unity packages like for Input, Spatial mapping etc. Utilities had started out simple but had incurred a lot of extra code which 90% of the feature folders would not find useful. This has been developer feedback for a while now. Please fix your PR to have the refactor I did with Common scripts separated out correctly under the Common namespace.

Common - has scripts that most feature areas would use or is a feature commonality in itself like anchor scripts. 

Utilities - On the other hand are all scripts that don't really depend on a feature area. Like FPS display, adaptive quality etc. They are more generic useful concepts which features don't have to use.

>>There are two materials still named HoloToolKit (and I don't have the energy at 2am to wander the project to fix them)
MixedRealityToolkit\Utilities\Materials\HoloToolkit_Default.mat
MixedRealityToolkit-Examples\Input\Materials\HoloToolkit_Focusable.mat
If someone else wants to pick those up after merge, please do.

The right way to approach this is to please file bugs so someone else could pick it up.

>>Polite request, no more merges in to dev until this is merged. Getting kinda fed up fixing the same files over and over again!

I completely respect the sentiment but unfortunately that does not seem like the right thing for other contributions. And hence this work has been so delayed due to the nature of its expanse. We could surely help expedite this PR once its ready with above changes.

**Comment by NeerajW** [comment]
>>That's why I raised that on the Issue proposal. I recommend renaming SpacialUnderstanding to SpacialUnderstandingManager in line with other areas. Shouldn't really have a namespace and a class of the same name.

Please let's not change the spelling for Spatial to Spacial.

**Comment by DDReaper** [comment]
Right, it might sound odd, but I'm going to restart this PR as the amount of changes seems to have broken GitHub.

@NeerajW Can you summarize in bullet points your issue, with the following exceptions.

- The Spatial rename was the correct spelling, I just miss typed in here :D 
- Taken your comments on board about Common / Utilities.  Whilst I don't completely agree, I respect your reasons for changing then in that way.
